### ISSUE TITLE
microk8s kubelet cannot connect to secret-operator node registrar

### ISSUE THREAD

**Issue by sheng-kenbun** [stackabletech/secret-operator#229]
### Affected version

zk-op 0.12.0

### Current and expected behavior

This happened in microk8s. After `apply -f zookeeper.yml` the operator created PVC, configmap and pods but no PV is created. The PVC and pods are in pending because they can not find PV to bind. The k8s `local-path-provisioner` is installed.

created PVC by the zookeeper operator:
```
data-kenbun-zk-server-default-0               Pending                       microk8s-hostpath        16m
kenbun-zk-server-default-0-client-tls-mount   Pending                       secrets.stackable.tech   16m
kenbun-zk-server-default-0-quorum-tls-mount   Pending                       secrets.stackable.tech   16m
```
but no created PV in the same namespace:
´´´
&> microk8s kubectl -n stackable get pv
No resources found
´´´

Expected:
Zookeeper pods are correctly started and running in microk8s, like in kind or minikube.

Is there already a known issue about this unexpected behaviour in microk8s? How can I get some further information about that why the operator failed to create pv? (operator logs, ect.)

### Additional context

commons=0.4.0, secret=0.6.0 and zookeeper=0.12.0 are installed by `stackablectl operator install`
```
NAME                                            READY   STATUS    RESTARTS   AGE
commons-operator-deployment-7bdd9fd6b7-2pt55    1/1     Running   0          10h
zookeeper-operator-deployment-bcdc889db-7wjqh   1/1     Running   0          10h
secret-operator-daemonset-njf5c                 3/3     Running   0          10h
```

### Environment

microk8s kubectl version --short:
```
Client Version: v1.25.4
Kustomize Version: v4.5.7
Server Version: v1.25.4
```
OS: ubuntu 22.04 x86_64

### Would you like to work on fixing this bug?

None

**Comment by nightkr** [comment]
The difference that  microk8s brings here is that it prefixes all kubelet paths with `/var/snap/microk8s/common`. This pushes the default path of `/var/lib/kubelet/plugins_registry/secrets.stackable.tech-reg.sock/secrets.stackable.tech-reg.sock` to become `/var/snap/microk8s/common/var/lib/kubelet/plugins_registry/secrets.stackable.tech-reg.sock/secrets.stackable.tech-reg.sock`, which is 122 characters.

The problem is that `connect(AF_UNIX)` only allows paths that are up to 108 characters long.

You can work around this by running `kubectl patch ds/secret-operator-daemonset -p '{"spec":{"template":{"spec":{"volumes":[{"name":"registration-sock","hostPath":{"path":"/var/lib/kubelet/plugins_registry"}}]}}}}'` . I'll also submit a PR in a moment to fix this by default.

After running the command you will need to either `rmdir /var/snap/microk8s/common/var/lib/kubelet/plugins_registry/secrets.stackable.tech-reg.sock` or reboot your machine for secret-operator to start back up correctly.

**Comment by nightkr** [comment]
I spoke too soon, we're also unable to mount anything, since Microk8s passes the "full" path (including /var/snap) to the CSI node driver, but our driver mounts assume that the kubelet's state dir is in /var/lib/kubelet. The investigation continues...

**Comment by nightkr** [comment]
After #232 the operator should now start correctly. However, zookeeper-operator is now reporting some networking issues on microk8s, I'll create a separate issue for that... Sorry for the confusion!
